src/test/java/net/masterthought/cucumber/generators/integrations/FeaturesOverviewPageIntegrationTest.java

```diff
@@ -149,6 +150,38 @@ public void generatePage_generatesStatsTableBody() {
         secondRow.getReportLink().hasLabelAndAddress("Second feature", "report-feature_1_1515379431.html");
     }
 
+    @Test
+    public void generatePage_WithSpecifiedQualifier_generatesStatsTableBody() {
+
+        // given
+        String testQualifier = "testQualifier";
+        configuration.setQualifier(FilenameUtils.removeExtension(SAMPLE_JSON), testQualifier);
```

Comment:
I would expect https://github.com/damianszczepanik/cucumber-reporting/pull/936/files#diff-87720169362dbe935db6e3bfbb83b6a6R311 takes filename WITH extension while here looks like without

Reply:
No, the filename is taken without the extension.  When the qualifier is retrieved in `ReportParser.java` lines 95-97, it uses the result of `extractQualifier(jsonFile)` as the key to get the qulifier - and that method strips out the extension.

Reply:
No, filename is expected to be without the extension, as can be seen in `ReportParser.java` lines 95-97.  Results of call to `extractQualifier(jsonFile)` is used as the key for lookup and that method strips `.json` extension.